homeassistant/components/husqvarna_automower/device_tracker.py

```diff
@@ -44,9 +48,9 @@ def source_type(self) -> SourceType:
     @property
     def latitude(self) -> float:
         """Return latitude value of the device."""
-        return self.mower_attributes.positions[0].latitude
+        return cast(list[Positions], self.mower_attributes.positions)[0].latitude
```

Comment:
Why is a cast needed here?

Reply:
The Positions list is optional. Because of that it's defined in the library as `list[Positions] | None`.
As the entity is only added, when there are positions, I have to tell mypy that it is not `None`

Reply:
Usually we guard an assert that it's not `None` with `if TYPE_CHECKING` so it's clear it's not missing typing or wrong in the library